sentiment_analysis.ipynb

```diff
@@ -0,0 +1,258 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# **IMPORTANT**\n",
+    "To run this notebook, you need to download the ThirdAI docker container by signing up [here](https://www.thirdai.com/try-bolt/)."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## **Sentiment analysis with BOLT**\n",
+    "\n",
+    "We will walk through the process of building a sentiment analysis model with BOLT from data preprocessing all the way to inference. This notebook is structured as follows:\n",
+    "1. Selecting and preprocessing the dataset\n",
+    "2. Defining the BOLT network\n",
+    "3. Training the network\n",
+    "4. Inference"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### **1. Choosing and preprocessing the dataset**\n",
+    "At our webinar on April 6th, we showed how BOLT reached state-of-the-art accuracy on the [Yelp Reviews](https://github.com/huggingface/datasets/blob/master/datasets/yelp_polarity/yelp_polarity.py) dataset and demonstrated that a model trained on the [Amazon Polarity](https://huggingface.co/datasets/amazon_polarity) dataset can be used for interactive, real-time sentiment analysis. Now, we want to give you a chance to try out BOLT with a dataset of your choice. \n",
+    "\n",
+    "We provided a utility function that converts text datasets into input vectors and saves them in SVM format. The text dataset must be a CSV file where each row follows this format:\n",
+    "\n",
+    "\\<pos or neg\\>,\\<text\\> \n",
+    "\n",
+    "For example, we can have a training corpus called example_train.csv that contains the following:\n",
+    "```\n",
+    "pos,Had a great time at the webinar.\n",
```

Comment:
Can you say something about how many training and testing examples we need? From this I would think 2 was enough